src/ecdsa/mod.rs

```diff
@@ -0,0 +1,127 @@
+use std::convert::TryInto;
+
+use ecdsa::{
+    elliptic_curve::{sec1::ToEncodedPoint, Field},
+    hazmat::SignPrimitive,
+};
+use message_digest::MessageDigest;
+use serde::{Deserialize, Serialize};
+use tracing::error;
+
+use crate::{
+    crypto_tools::{k256_serde, message_digest, rng},
+    sdk::api::{BytesVec, TofnFatal, TofnResult},
+};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct KeyPair {
+    signing_key: k256_serde::SecretScalar,
+    verifying_key: k256_serde::ProjectivePoint,
+}
+
+impl KeyPair {
+    /// SEC1-encoded compressed curve point
+    pub fn encoded_verifying_key(&self) -> TofnResult<[u8; 33]> {
+        // TODO make this work with k256_serde::ProjectivePoint::to_bytes
+        self.verifying_key
+            .as_ref()
+            .to_affine()
+            .to_encoded_point(true)
+            .as_bytes()
```

Comment:
Same thing: `try_from` and `try_into` both return a `Result`.

Worse still: I could return a `&[u8]` instead of a freshly-allocated `Vec<u8>` except all the `to_bytes` methods encode on the fly and so `&[u8]` can't compile because it's a reference to temporary data. 😞

Reply:
Hmm, I thought the issue was how about how to make use of `to_bytes` in `encoded_verifying_key`. But it seems like your concern is to avoid having a `TofnResult` return type instead of just `[u8; 33]` on the `encoded_verifying_key` method? Tony's suggestion below should help with the latter it seems.